binary_search_tree/BinarySearchTree.py

```diff
@@ -0,0 +1,179 @@
+# Node for Binary Search Tree
+class Node:
+
+    def __init__(self, data):
+        self.data = data
+        self.left = None
+        self.right = None
+
+# function for find the left side right most node in binary search tree
+def find_leftside_rightmost(node):
+    left_side_rightmost = node.left
+    prev_node = node
+    while left_side_rightmost.right is not None:
+        prev_node = left_side_rightmost
+        left_side_rightmost = left_side_rightmost.right
+    return left_side_rightmost, prev_node
+
+
+class BST:
+
+    def __init__(self):
+
+        self.root = None    # Root for Binary Search Tree
+        self.count = None   # Size of Binary Search Tree
+
+    def insert(self, data):
+
+        new_node = Node(data)   # Create a node for the new entry
+
+        if self.root is None:
+            self.root = new_node
+
+        else:
+
+            cur_node = self.root
+            prev_node = None
+
+            while cur_node is not None:
+
+                prev_node = cur_node
+                if data <= cur_node.data:
+                    cur_node = cur_node.left
+                else:
+                    cur_node = cur_node.right
+
+            if data <= prev_node.data:
+                prev_node.left = new_node
+            else:
+                prev_node.right = new_node
+
+    def search(self, data):    # Print a statement according to the search result
+
+        cur_node = self.root
+        while cur_node is not None and data != cur_node.data:
+            if data < cur_node.data:
+                cur_node = cur_node.left
+            else:
+                cur_node = cur_node.right
+
+        # if node found, cur_node not to be None
+        if cur_node is None:
+            print("Invalid Search")
+        else:
+            print("Node was found successfully.")
+
+    # Function for deleting node using parent node
+    def delete_node(self, del_node, parent_node):
+
+        isLeft = False    # True if node is in left side of the parent, else it's in right side
+
+        if parent_node is None:   # For root node deleting
+
+            # Root hasn't children
+            if del_node.left is None and del_node.right is None:
+                self.root = None
+
+            # Root has a child
+            elif del_node.left is None or del_node.right is None:
+                if del_node.left is None:
+                    self.root = del_node.right
+                else:
+                    self.root = del_node.left
+
+            # Root has 2 children
+            else:
+                left_side_rightmost, prev_node = find_leftside_rightmost(del_node)
```

Comment:
The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `pep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpc9jm3hgg/binary_search_tree/BinarySearchTree.py
+++ b/tmp/tmpc9jm3hgg/binary_search_tree/BinarySearchTree.py
@@ -83,7 +83,8 @@
 
             # Root has 2 children
             else:
-                left_side_rightmost, prev_node = find_leftside_rightmost(del_node)
+                left_side_rightmost, prev_node = find_leftside_rightmost(
+                    del_node)
                 self.root.data = left_side_rightmost.data
                 if del_node == prev_node:
                     prev_node.left = None
```